css-properties-values-api/Overview.bs

```diff
@@ -601,6 +601,125 @@ referenced, otherwise the declaration is
+
+	<pre class="prod def" nohighlight>
+	@property <<custom-property-name>> {
+		<<declaration-list>>
+	}
+	</pre>
+
+The ''@property'' rule accepts the descriptors defined in this chapter.
+
+Issue(845): What happens when we have both an ''@property'' rule, and a
+{{registerProperty()}} call for the same property?
+
+Issue(846): What happens when ''@property'' appears inside a shadow tree?
+
+The 'syntax' descriptor {#syntax-descriptor}
+--------------------------------------------
+
+	<pre class='descdef'>
+	Name: syntax
+	Value: <<string>>
+	For: @property
+	Initial: "*"
+	</pre>
+
+Specifies the syntax of the custom property, on the form defined by
+[[#supported-syntax-strings]]. This descriptor is equivalent to the
+{{PropertyDescriptor/syntax|syntax}} member of {{PropertyDescriptor}}.
+
+The 'inherits' descriptor {#inherits-descriptor}
+------------------------------------------------
+
+	<pre class='descdef'>
+	Name: inherits
+	Value: true | false
+	For: @property
+	Initial: Nothing
+	</pre>
+
+Specifies whether or not the custom property inherits. This is equivalent to the
+{{PropertyDescriptor/inherits}} member of {{PropertyDescriptor}}.
+
+This descriptor is required: if omitted, the ''@property'' rule will not
+register the property.
+
+The 'initial-value' descriptor {#initial-value-descriptor}
+----------------------------------------------------------
+
+	<pre class='descdef'>
+	Name: initial-value
+	Value: <<declaration-value>>
+	For: @property
+	Initial: Nothing
+	</pre>
+
+Specifies the [[css-cascade-4#initial-values|initial value]] of the custom
+property. This is equivalent to the {{PropertyDescriptor/initialValue}} member
+of {{PropertyDescriptor}}.
+
+This descriptor may be omitted when the value of the 'syntax' descriptor is
+<code>"*"</code>, in which case the initial value is
+[[css-variables#invalid-variables|the empty value invalid for substituion]]
+described by [[css-variables]]. Otherwise, the 'initial-value' descriptor is
+required, and the ''@property'' rule will not register the property if it's
+omitted.
+
+The value specified for 'initial-value' must be [=computationally independent=],
+and match the grammar corresponding to the specified 'syntax'.
+
+Extensions to the <code>CSSRule</code> interface {#extensions-to-css-rule-interface}
+------------------------------------------------------------------------------------
+
+The <code>CSSRule</code> interface is extended as follows:
+
+<pre class='idl'>
+partial interface CSSRule {
+    const unsigned short PROPERTY_RULE = 16;
+};
+</pre>
+
+The <code>CSSPropertyRule</code> interface {#the-css-property-rule-interface}
+-----------------------------------------------------------------------------
+
+The <code>CSSPropertyRule</code> interface represents an ''@property'' rule.
+
+<pre class='idl' export>
+[Exposed=Window]
+interface CSSPropertyRule : CSSRule {
+    attribute CSSOMString name;
+    readonly attribute CSSStyleDeclaration style;
```

Comment:
why not have the descriptors as bare attributes here?

Reply:
+1, CSSStyleDeclaration has a couple interop problems (WebKit and blink define all CSS properties within it, so you can have `'width' in fontFaceRule.style` return true, which is broken IMO) which I'd rather not repeat.

Reply:
:+1: Will do.